invenio_drafts_resources/services/draft.py

```diff
@@ -71,85 +62,138 @@ def draft_cls(self):
         """Factory for creating a record class."""
         return self.config.draft_cls
 
-    @property
-    def draft_data_validator(self):
-        """Returns an instance of the draft data validator."""
-        return self.config.draft_data_validator
-
-    @property
-    def draft_resolver(self):
-        """Factory for creating a draft resolver instance."""
-        return self.config.resolver_cls(
-            pid_type=self.config.resolver_pid_type,
-            getter=self.draft_cls.get_record,
-            registered_only=False
-        )
-
-    def resolve_draft(self, id_):
-        """Resolve a persistent identifier to a record draft."""
-        return self.draft_resolver.resolve(id_)
-
     # High-level API
     # Inherits record read, search, create, delete and update
 
     def read_draft(self, id_, identity):
         """Retrieve a record."""
-        pid, record = self.resolve_draft(id_)
-        self.require_permission(identity, "read", record=record)
+        pid, draft = self.pid_manager.resolve(id_, draft=True)
+        self.require_permission(identity, "read", record=draft)
         # Todo: how do we deal with tombstone pages
-        return self.resource_unit(pid=pid, record=record)
+        return self.resource_unit(pid=pid, record=draft, links=None)
+
+    def update_draft(self, id_, data, identity):
+        """Replace a record."""
+        # TODO: etag and versioning
+        pid, draft = self.pid_manager.resolve(id_, draft=True)
+        # Permissions
+        self.require_permission(identity, "update", record=draft)
+        validated_data = self.data_validator.validate(data, partial=True)
+        # FIXME: extract somewhere else
+        self._patch_data(draft, validated_data)
+        draft.update(draft.dumps())
+        draft.commit()
+
+        if self.indexer:
+            self.indexer.index(draft)
+
+        return self.resource_unit(pid=pid, record=draft, links=None)
 
     def create(self, data, identity):
         """Create a draft for a new record.
 
         It does not eagerly create the associated record.
         """
         self.require_permission(identity, "create")
-        validated_data = self.draft_data_validator.validate(data)
+        validated_data = self.data_validator.validate(data, partial=True)
         rec_uuid = uuid.uuid4()
-        pid = self.minter(record_uuid=rec_uuid, data=validated_data)
+        pid = self.pid_manager.mint(record_uuid=rec_uuid, data=validated_data)
```

Comment:
:100:  Much better than the minters that we used to have.

Reply:
😃